src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1786,12 +1786,22 @@ void CodeGen::genAvxFamilyIntrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_AVX512F_ConvertToVector256Int32:
```

Comment:
why was this moved up?

Reply:
To be consistent with the other places that specially handle this one path and to avoid the cost of the `varTypeIsFloating` check for the many intrinsics where it can't be true.